Source/Widgets/Geocoder/GeocoderViewModel.js

```diff
@@ -53,33 +64,123 @@ define([
         }
         //>>includeEnd('debug');
 
+        if (defined(options.geocoderServices)) {
+            this._geocoderServices = options.geocoderServices;
+        } else {
+            this._geocoderServices = [
+                new CartographicGeocoderService(),
+                new BingMapsGeocoderService()
+            ];
+        }
+
+        var errorCredit;
         this._url = defaultValue(options.url, 'https://dev.virtualearth.net/');
         if (this._url.length > 0 && this._url[this._url.length - 1] !== '/') {
             this._url += '/';
         }
 
         this._key = BingMapsApi.getKey(options.key);
-        var errorCredit = BingMapsApi.getErrorCredit(options.key);
+        this._defaultGeocoderOptions = {
+            url: this._url,
+            key: this._key
+        };
+
+        if (defined(options.key)) {
+            errorCredit = BingMapsApi.getErrorCredit(options.key);
+        }
         if (defined(errorCredit)) {
             options.scene._frameState.creditDisplay.addDefaultCredit(errorCredit);
         }
 
+        this._viewContainer = options.container;
         this._scene = options.scene;
         this._flightDuration = options.flightDuration;
         this._searchText = '';
         this._isSearchInProgress = false;
-        this._geocodeInProgress = undefined;
+        this._geocodePromise = undefined;
         this._complete = new Event();
+        this._suggestions = [];
+        this._selectedSuggestion = undefined;
+        this._showSuggestions = true;
+        this._updateCamera = updateCamera;
+        this._adjustSuggestionsScroll = adjustSuggestionsScroll;
+        this._updateSearchSuggestions = updateSearchSuggestions;
+        this._handleArrowDown = handleArrowDown;
+        this._handleArrowUp = handleArrowUp;
 
         var that = this;
+
+        this._suggestionsVisible = knockout.pureComputed(function () {
+            var suggestions = knockout.getObservable(that, '_suggestions');
```

Comment:
My knockout-es5 is a little rusty, but is this `knockout.getObservable` needed here?  I think you can just use `that._suggestions` directly and everything will work.  CC @shunter